04-onion-routing.md

```diff
@@ -367,6 +368,102 @@ otherwise meets the amount criterion (eg. some other failure, or
 invoice timeout), however if it were to fulfill only some of them,
 intermediary nodes could simply claim the remaining ones.
 
+### Onion Messages
+
+Onion messages have an onion with an alternate `hop_payload`
+format: a `bigsize` followed by a `onionmsg_payload`.  Note that there
```

Comment:
Please, no bigsize, just drop it entirely and use the 2 byte length in the `onion_message` struct.

Reply:
This matches the modern onion format in htlc onions, which means no fresh parser to write.

Reply:
Huh, its a fresh parser either way because a completely different TLV namespace? I guess your length read is at a different layer from the fields read, then?

Reply:
Yes, we have wrap/unwrap routines, which just needed to be rejigged to allow a size param (rather than 1300 hardcoded).  Was trying to KISS...